client/app/pages/queries/utils/deleteQueryVisualization.js

```diff
@@ -0,0 +1,15 @@
+import { clone, extend, filter } from "lodash";
+import { Visualization } from "@/services/visualization";
+import notification from "@/services/notification";
+
+export default function deleteQueryVisualization(query, visualization) {
+  return Visualization.delete({ id: visualization.id })
+    .$promise.then(() => {
+      const filteredVisualizations = filter(query.visualizations, v => v.id !== visualization.id);
+      return Promise.resolve(extend(clone(query), { visualizations: filteredVisualizations }));
```

Comment:
While working on getredash/redash#4468 I realized that `_.clone` is not enough to properly clone query instance, so I added a `.clone()` method to Query service. No need to fix it in this PR, but we should keep this in mind and fix in one of  subsequent PRs